nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-cluster/src/main/java/org/apache/nifi/cluster/manager/impl/WebClusterManagerCoordinator.java

```diff
@@ -0,0 +1,164 @@
+        final Node node = clusterManager.getRawNode(nodeId.getId());
+
+        if (node == null) {
+            final ConnectionRequest connectionRequest = new ConnectionRequest(nodeId);
+            clusterManager.requestConnection(connectionRequest);
+        } else {
+            node.setStatus(Status.DISCONNECTED);
+            clusterManager.requestReconnection(nodeId.getId(), "Anonymous");
+        }
+    }
+
+    @Override
+    public void finishNodeConnection(final NodeIdentifier nodeId) {
+        final Node node = clusterManager.getRawNode(nodeId.getId());
+        if (node == null) {
+            logger.error("Attempting to Finish Node Connection but could not find Node with Identifier {}", nodeId);
+            return;
+        }
+
+        node.setStatus(Status.CONNECTED);
+    }
+
+    @Override
+    public void requestNodeDisconnect(final NodeIdentifier nodeId, final DisconnectionCode disconnectionCode, final String explanation) {
+        try {
+            clusterManager.requestDisconnection(nodeId, false, explanation);
+
+            if (disconnectionCode == DisconnectionCode.LACK_OF_HEARTBEAT) {
+                final Node node = clusterManager.getRawNode(nodeId.getId());
+                if (node != null) {
+                    node.setHeartbeatDisconnection();
+                }
+            }
+        } catch (final Exception e) {
+            logger.error("Failed to request node {} disconnect from cluster due to {}", nodeId, e);
+            logger.error("", e);
+        }
+    }
+
+    @Override
+    public void disconnectionRequestedByNode(final NodeIdentifier nodeId, final DisconnectionCode disconnectionCode, final String explanation) {
+        final Node node = clusterManager.getRawNode(nodeId.getId());
+        if (node != null) {
+            node.setStatus(Status.DISCONNECTED);
+
+            final Severity severity;
+            switch (disconnectionCode) {
+                case STARTUP_FAILURE:
+                case MISMATCHED_FLOWS:
+                case UNKNOWN:
+                    severity = Severity.ERROR;
+                    break;
+                default:
+                    severity = Severity.INFO;
+                    break;
+            }
+
+            reportEvent(nodeId, severity, "Node disconnected from cluster due to " + explanation);
+        }
+    }
+
+    @Override
+    public NodeConnectionStatus getConnectionStatus(final NodeIdentifier nodeId) {
+        final Node node = clusterManager.getNode(nodeId.getId());
+        if (node == null) {
+            return null;
+        }
+
+        final Status status = node.getStatus();
+        final NodeConnectionState connectionState = NodeConnectionState.valueOf(status.name());
+        return new NodeConnectionStatus(connectionState, node.getConnectionRequestedTimestamp());
+    }
+
+    @Override
+    public Set<NodeIdentifier> getNodeIdentifiers(final NodeConnectionState state) {
```

Comment:
Not a big deal, but personally I would probably name the method 'extract/build....' since 'get' conventionally implies something that exists.